### PR TITLE
[WIP] Redesigned Start Center prototype

### PR DESCRIPTION
**Pull request by dmitrio95**
This is the prototype of the redesigned Start Center implemented in QML. Current implementation uses the existing QML plugins framework to interact with the main application although that framework still needs some expansions and extra QML bindings. The related changes are also included to this branch.

In order to run the current version the new Start Center prototype a "StartCenterDemo" plugin should be activated in Plugin Manager, then it will be available via Plugins→Start Center menu entry.

This implementation is far from being complete, and aside of the current feature-incompleteness contains some possible layout implementation issues and lacks a decent support for keyboard navigation. These issues are still to be addressed within this Start Center reworking process.

### CODE REVIEW

**Comment by anatoly-os** [comment]
The prototype example, btw.
Screenshot 2019-05-17 at 11 27 29

File: mscore/plugin/api/qmlpluginapi.h [hunk 1]
```diff
@@ -83,6 +83,11 @@ class PluginAPI : public Ms::QmlPlugin {
       /** current score, if any (read only) */
       Q_PROPERTY(Ms::PluginAPI::Score* curScore     READ curScore)
 //TODO-ws      Q_PROPERTY(QQmlListProperty<Ms::Score> scores READ scores)
+      /**
```
Comment (Jojo-Schmitz) [inline mscore/plugin/api/qmlpluginapi.h]: this here will probably cause a merge conflict with #5021, but should be easy to fix

**Comment by RobFog** [comment]
What is the redesign trying to solve?

**Comment by anatoly-os** [comment]
@RobFog the idea is to move everything which doesn't relate to creating scores to the start center like Extensions, Plugins, Save Online, etc. Also, it should make starting experience smoother. Also, there will be an embedded help center.

I will prepare a post proposing the changes and the purpose of the redesign.

Btw, this is a start of global improvements in the visual and interface design in MuseScore.

**Comment by MarcSabatella** [comment]
I'll be very interested to discuss the help center!

**Comment by anatoly-os** [comment]
The development of the Start Center refinement is on hold.

**Pull request closed without merging**